### ISSUE TITLE
Mesh Pipe

### ISSUE THREAD

**Issue by petrasvestartas** [CGAL/cgal#6292]
Hi,

I would like to ask if there any CGAL method or project that allows to create a pipe around a polyline?
I could do it myself, but it seems that this a very common case that could be made my many people before me.
If yes, could you point me to any example file.
If not I try to do this myself using polygon orientation on planes.

![image](https://user-images.githubusercontent.com/18013985/152011088-f0901644-31e0-47cc-a71f-ef15eaacfbb3.png)

**Comment by lrineau** [comment]
@CGAL/geometryfactory Do we not have a function, somewhere in the 3D demo, that allows to extrude a mesh along a curve?

**Comment by sloriot** [comment]
There used to be a branch but the code was not good enough and it was dropped. I do have a working version somewhere but I'd need some time to find it and put it inside a function.

**Comment by petrasvestartas** [comment]
Dear @lrineau  and @sloriot ,

Thank you for your replies.
It would be really nice have an example.

**Comment by petrasvestartas** [comment]
@sloriot 
Any update if the code is available?

**Comment by sloriot** [comment]
Sorry I don't have time for now.

**Comment by sloriot** [comment]
See #6317 which is working only on `Surface_mesh`. The code still need some works but at least it works.

**Comment by petrasvestartas** [comment]
Thank you @sloriot and @GilesBathgate 

Dear @GilesBathgate  do you have a test example with a simple data-set of a polyline ? 
I assume I need to use all the file with necessary headers to run these two methods. Is it correct?

**Comment by GilesBathgate** [comment]
> Dear @GilesBathgate do you have a test example with a simple data-set of a polyline ?

No. It doesn't work with a polyline, only a height, or a radius.

>I assume I need to use all the file with necessary headers to run these two methods. Is it correct?

Yeah, it has a dependency on most of the rest of the project. You probably won't be able to use standalone. I provided it only for reference. Essentially what I do is find the perimeter of the source mesh (2d faces in 3d space), I find the normal of this perimeter using `CGAL::normal_vector_newell_3` then I build sides and endcaps using the perimeter, making sure that the endcaps have the right winding with respect to the normal.

>And do I need to use the latest cgal version?

No, it compiles against quite old CGAL versions.
